### PR TITLE
Avoid exception on startup with JDK 9

### PR DESCRIPTION
**Pull request by avdv**
This PR mitigates the effect of using the ensime SBT plugin with JDK 9.

It by no means makes ensime usable with JDK 9. See #367

### CODE REVIEW

File: src/main/scala/EnsimePlugin.scala [hunk 1]
```diff
@@ -242,11 +242,15 @@ object EnsimePlugin extends AutoPlugin {
       }
     }
 
+  private val isPreJava9 = sys.props("java.version").contains(".")
```
Comment (fdeitylink) [inline src/main/scala/EnsimePlugin.scala]: With Java `9.0.1` this wouldn't work, so checking if the version string starts with `1` or comparing the major version against `9` should work.

Comment (avdv) [reply src/main/scala/EnsimePlugin.scala]: > With Java 9.0.1 this wouldn't work, so checking if the version string starts with 1 or comparing the major version against 9 should work.

Yes, but your logic will then fail with Java 10 again...

I changed the code to use `java.specification.version`, which should work correctly according to http://openjdk.java.net/jeps/223

**Comment by amanjpro** [comment]
Hi

**Comment by fommil** [comment]
my preferred solution for this would be to move the calculation of the `tools.jar` into a `Setting` that returns `Seq[File]`, with the java9+ detection in there, and allowing user override.

ensime 3.0 removed the debugger, which means we might not even need it in the future.

Can you please update your PR?

**Comment by avdv** [comment]
> Can you please update your PR?

Sure thing. Please have a look...

**Review by fommil — APPROVED** [review]

**Comment by fommil** [comment]
thanks! I like this. I'll merge if the travis / appveyor tests pass, which at least gives us regression testing on java 8. I assume you've manually tried it on java 9?

**Comment by fommil** [comment]
@avdv I'm releasing v2.5.0, thanks for looking into this.

**Comment by fommil** [comment]
umm, but it seems that sbt fails to start up due to

```
sbt
expr: syntax error: unexpected argument ‘1.6’

No Java Development Kit (JDK) installation was detected.
Please go to http://www.oracle.com/technetwork/java/javase/downloads/ and download.
```

**Comment by fommil** [comment]
hacknig the sbt script to remove the `javaCheck` lets me get past this

**Pull request merged**